### PR TITLE
[PD-3278] add timespan based options

### PR DESCRIPTION
**Pull request by mwarnsleyDE**
## What's this PR do?
PR adds the timeline chart into Analytics

## Screenshots (if appropriate)
screen shot 2017-06-01 at 11 49 15 am

## Where should the reviewer start?
Start inside of any of the analytics reports

## How should this be manually tested?
Pull down the repository. Go into any of the analytics apps and click on the day, week, or month field and you should see the timeline drop down with the data inside of it

## Any background context you want to provide?
Will need some extensive testing to make sure the timeline and data coming back are correct

### CODE REVIEW

File: gulp/src/analytics/components/Charts/Line/LineChart.js [hunk 1]
```diff
@@ -1,35 +1,72 @@
...
-          {name: 'Page D', uv: 2780, pv: 3908, amt: 2000},
-          {name: 'Page E', uv: 1890, pv: 4800, amt: 2181},
-          {name: 'Page F', uv: 2390, pv: 3800, amt: 2500},
-          {name: 'Page G', uv: 3490, pv: 4300, amt: 2100},
-    ];
+    const {width, height, lineData} = this.props;
+    const getDataKeys = Object.keys(lineData[0]);
+    let xAxisKey;
+    let lineKey;
+    if (getDataKeys.indexOf('day') > -1 && getDataKeys.indexOf('Job Views') > -1) {
```
Comment (aljordan82) [inline gulp/src/analytics/components/Charts/Line/LineChart.js]: I would extract this line as a variable
`const someVar =  getDataKeys.indexOf('Job Views') > -1`

File: gulp/src/analytics/actions/header-actions.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import {errorAction} from '../../common/actions/error-actions';
```
Comment (aljordan82) [inline gulp/src/analytics/actions/header-actions.js]: Seems like an unintentional file update.

File: gulp/src/analytics/actions/header-actions.js [hunk 3]
```diff
@@ -3,17 +3,24 @@ import {errorAction} from '../../common/actions/error-actions';
...
       const currentReport = getState().pageLoadData.activeReport;
       const storedDates = {
         startDate: getState().pageLoadData.globalStartDate,
         endDate: getState().pageLoadData.globalEndDate,
       };
-      // Looping through the current tabs that are created and setting the stored filters of the ACTIVE tab that the user is currently on
-      const storedFilters = tabs.find((tab) => tab.active).activeFilters || [];
-      const exportResponse = await api.exportSelectedFilterData(storedFilters, currentReport, storedDates, presentationType);
 
+      for (let i = 0; i < tabs.length; i++) {
```
Comment (aljordan82) [inline gulp/src/analytics/actions/header-actions.js]: use` .find`

File: gulp/src/analytics/actions/header-actions.js [hunk 4]
```diff
@@ -3,17 +3,24 @@ import {errorAction} from '../../common/actions/error-actions';
...
       const link = document.createElement('a');
       const tabs = getState().pageLoadData.navigation;
       const currentReport = getState().pageLoadData.activeReport;
       const storedDates = {
         startDate: getState().pageLoadData.globalStartDate,
         endDate: getState().pageLoadData.globalEndDate,
       };
-      // Looping through the current tabs that are created and setting the stored filters of the ACTIVE tab that the user is currently on
-      const storedFilters = tabs.find((tab) => tab.active).activeFilters || [];
-      const exportResponse = await api.exportSelectedFilterData(storedFilters, currentReport, storedDates, presentationType);
```
Comment (aljordan82) [inline gulp/src/analytics/actions/header-actions.js]: Use .find instead of  the loop

File: gulp/src/analytics/actions/table-filter-actions.js [hunk 5]
```diff
@@ -87,3 +91,33 @@ export function doGetSearchData(searchValue) {
...
+export function doActivateTimeline(active) {
+  return async (dispatch, getState, {api, history}) => {
+    try {
+      dispatch(markNavLoadingAction(true));
+      const currentStartRange = getState().pageLoadData.globalStartDate;
+      const currentEndRange = getState().pageLoadData.globalEndDate;
+      const currentReport = getState().pageLoadData.activeReport;
+      const navigation = getState().pageLoadData.navigation;
+      let currentFilters;
+      for (let i = 0; i < navigation.length; i++) {
```
Comment (aljordan82) [inline gulp/src/analytics/actions/table-filter-actions.js]: Use `.find `

File: gulp/src/analytics/components/Charts/Line/LineChart.js (hunk 1 above)
Comment (aljordan82) [inline gulp/src/analytics/components/Charts/Line/LineChart.js]: Please add comments to this switch statment.

File: gulp/src/analytics/actions/header-actions.js (hunk 4 above)
Comment (mwarnsleyDE) [inline gulp/src/analytics/actions/header-actions.js]: @aljordan82 is there a specific reason you want to use .find instead. A standard for loop is faster performance wise compared to .find

Comment (aljordan82) [inline gulp/src/analytics/actions/header-actions.js]: It's just easier to read. The performance is not an issue here.

File: gulp/src/analytics/actions/table-filter-actions.js (hunk 5 above)
Comment (mwarnsleyDE) [inline gulp/src/analytics/actions/table-filter-actions.js]: @aljordan82 fixed

File: gulp/src/analytics/actions/header-actions.js (hunk 4 above)
Comment (mwarnsleyDE) [inline gulp/src/analytics/actions/header-actions.js]: @aljordan82 fixed

File: analytics/api.py [hunk 6]
```diff
@@ -63,6 +72,90 @@ def validate_analytics_query_request(query_data):
...
+            view_type: sum(d['views'] for d in timeline_data[i: i+7])
+        })
+    for i in range(0, days, 30):
+        prepared_timeline_data['by_month'].append({
+            'month': timeline_data[i+30 < days and i+30 or days-1]['day'].isoformat(),
+            view_type: sum(d['views'] for d in timeline_data[i: i+30])
+        })
+
+    # # fill data for each month - not every 30 days but by calendar months
+    # if start_date_mdy[0] < end_date_mdy[0]:
```
Comment (almathew) [inline analytics/api.py]: Commented out code?

File: gulp/src/analytics/components/Header/ExportOptions.js [hunk 7]
```diff
@@ -38,6 +38,14 @@ class ExportOptions extends Component {
       exportCSV: false,
     });
   }
+  // exportReport() {
```
Comment (almathew) [inline gulp/src/analytics/components/Header/ExportOptions.js]: @mwarnsleyDE commented out code

**Comment by almathew** [comment]
LGTM now that all the commented out code is gone.

**Pull request merged**